test/integration_test.py

```diff
@@ -1035,3 +1044,66 @@ def test_local_filtered_index_endpoints(self):
                                              catalog=self.catalog)).url
         response = requests.get(url)
         self.assertEqual(200, response.status_code)
+
+
+class CanBundleScriptIntegrationTest(AzulTestCase):
+
+    def _test_catalog(self, catalog: config.Catalog):
+        with self.subTest(catalog=catalog.name,
+                          repository=catalog.plugins['repository']):
+            fqid = self.bundle_fqid(catalog.name)
+            with tempfile.TemporaryDirectory() as d:
+                exit_code = self._can_bundle(source=str(fqid.source.spec),
+                                             uuid=fqid.uuid,
+                                             version=fqid.version,
+                                             output_dir=d)
+                self.assertEqual(exit_code, 0)
+                parts = ['manifest', 'metadata']
+                expected_files = sorted(f'{fqid.uuid}.{part}.json' for part in parts)
+                generated_files = sorted(os.listdir(d))
+                self.assertListEqual(generated_files, expected_files)
+
+    def test_can_bundle_configured_catalogs(self):
+        for catalog_name, catalog in config.catalogs.items():
+            if catalog.is_integration_test_catalog:
+                self._test_catalog(catalog)
+
+    canned_repo = 'https://github.com/HumanCellAtlas/schema-test-data/tree/master/tests'
+
+    @mock.patch.dict(os.environ,
+                     azul_canned_sources=f'{canned_repo}:')
+    @mock.patch('azul.Config.catalogs', new_callable=PropertyMock)
+    def test_can_bundle_canned_repository(self, mock_config_catalogs):
+        mock_catalog_name = 'testcanned'
+        mock_catalog = config.Catalog(name=mock_catalog_name,
+                                      atlas='hca',
+                                      plugins={
+                                          'metadata': 'hca',
+                                          'repository': 'canned'
+                                      })
+        mock_config_catalogs.return_value = {mock_catalog_name: mock_catalog}
+        self._test_catalog(mock_catalog)
+
+    def _can_bundle(self,
+                    source: str,
+                    uuid: str,
+                    version: str,
+                    output_dir: str
+                    ) -> int:
+        args = [
+            'python3',
+            f'{config.project_root}/scripts/can_bundle.py',
+            '--source', source,
+            '--uuid', uuid,
+            '--version', version,
+            '--output-dir', output_dir
+        ]
+        args = [shlex.quote(arg) for arg in args]
+        result = subprocess.run(args)
```

Comment:
Why not just import and call `main()` directly? It would make debugging easier in the case of a failure.

Reply:
I thought we weren't supposed to import from `scripts/` from other locations

Reply:
Now that you mention it, I remember hearing that too... though I wonder if testing a script like this is worthwhile exception.

Reply:
I think we have to make it an exception because `mock.patch` can't cross process boundaries, and we need to patch the configured catalogs in order to test the canned repository.

Reply:
(Worth noting that the Pycharm debugger *does* seem to be capable of crossing process boundaries. When I put a breakpoint in the script and invoked it via `subprocess.run`, it stopped at the breakpoint).

Reply:
> (Worth noting that the Pycharm debugger does seem to be capable of crossing process boundaries. When I put a breakpoint in the script and invoked it via subprocess.run, it stopped at the breakpoint).

Good to know, thanks for testing this.

Reply:
That can be turned on or off. 

![image](https://user-images.githubusercontent.com/5143256/125994442-2f3db7a0-4f95-4b6f-ac68-01127e023798.png)

Note `load_script` in azul.modules.